slash_commands/adora-sac.js

```diff
@@ -134,17 +137,36 @@ async function execute(interaction) {
 
         let resultCost = 0;
         let result = [];
-        for (let i = moneyToSpendDiv5; i <= dpSize; ++i) {
-            if (bagSize[i] !== Infinity) {
-                resultCost = i * 5;
-                while (i > 0) {
-                    result.push(costsDiv5ToUpgrade.get(lastItem[i]));
-                    i -= lastItem[i];
+
+        for (let startingMoneyDiv5 = moneyToSpendDiv5; startingMoneyDiv5 <= dpSize; ++startingMoneyDiv5) {
```

Comment:
We might want to adjust this so that we check amounts slightly below the target money to spend as they can sometimes be optimal with the tower limit.

Reply:
@Landmaster Oh yeah that makes sense. Technically, this is true now if the user enters an XP amount where `xp % 20 <= 4` because spending the raw $4 is more efficient than spending the extra $5.